contracts/registries/LicenseRegistry.sol

```diff
@@ -193,14 +193,23 @@ contract LicenseRegistry is ILicenseRegistry, AccessManagedUpgradeable, UUPSUpgr
             revert Errors.LicensingModule__LicenseTermsNotFound(licenseTemplate, licenseTermsId);
         }
 
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        if (_isExpiredNow(ipId)) {
+            revert Errors.LicenseRegistry__IpExpired(ipId);
+        }
+
+        if (_hasIpAttachedLicenseTerms(ipId, licenseTemplate, licenseTermsId)) {
+            revert Errors.LicenseRegistry__LicenseTermsAlreadyAttached(ipId, licenseTemplate, licenseTermsId);
+        }
+
         if (_isDerivativeIp(ipId)) {
```

Comment:
Is this check redundant with the one in line 235? Derivatives will always have license terms attached

Reply:
oh, they are in different functions.  this check is for `attachLicenseTermsToIp` and LN 235 is for `registerDerivativeIp`. they won't be invoked within the same call.